sources/isal/igzip/adler32_base.h

```diff
@@ -0,0 +1,17 @@
+/*******************************************************************************
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+
+#ifndef ADLER32_BASE_H
+#define ADLER32_BASE_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+    uint32_t qpl_adler32_base(uint32_t adler32, uint8_t *start, uint32_t length);
```

Comment:
A c and c++-safe declaration of the function defined in `adler32_base.c` above. The missing declaration was automatically detected and automatically generated. (Note that an asm file declares that symbol `extern` here https://github.com/intel/qpl/blob/develop/sources/isal/igzip/igzip_multibinary.asm#L47 -- so name mangling must always be suppressed when a C++ compiler is used.)